internal/cmd/schema-gen/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	armadav1 "gitlab.com/nitrado/b2b/ec/armada/pkg/api/apis/armada/v1"
+	containerv1 "gitlab.com/nitrado/b2b/ec/armada/pkg/api/apis/container/v1"
+)
+
+type objInfo struct {
+	Obj      any
+	Filename string
+	FuncName string
+}
+
+var objs = []objInfo{
+	{
+		Obj:      armadav1.Resources{},
+		Filename: "ec/armada/schema_resources.go",
+		FuncName: "resourcesSchema",
+	},
+	{
+		Obj:      armadav1.EnvVar{},
+		Filename: "ec/armada/schema_env.go",
+		FuncName: "envSchema",
+	},
+	{
+		Obj:      &armadav1.Site{},
+		Filename: "ec/armada/schema_site.go",
+		FuncName: "siteSchema",
+	},
+	{
+		Obj:      &armadav1.Region{},
+		Filename: "ec/armada/schema_region.go",
+		FuncName: "regionSchema",
+	},
+	{
+		Obj:      &armadav1.Armada{},
+		Filename: "ec/armada/schema_armada.go",
+		FuncName: "armadaSchema",
+	},
+	{
+		Obj:      &armadav1.ArmadaSet{},
+		Filename: "ec/armada/schema_armadaset.go",
+		FuncName: "armadaSetSchema",
+	},
+	{
+		Obj:      &containerv1.Branch{},
+		Filename: "ec/armada/schema_branch.go",
+		FuncName: "branchSchema",
+	},
+}
```

Comment:
Is it normal that `schema_metadata.go` is missing? I assume it's imported by others or some such thing?

Reply:
It is a component that is used in all these, as a common schema.